lineapy/utils/config.py

```diff
@@ -30,50 +37,64 @@ class lineapy_config:
     replace with values in environmental variables if possible. Finally, it sets
     all values in environmental variables.
 
+    :param home_dir: home directory of LineaPy (must be local)
+    :param database_url: database connection string for LineaPy database
+    :param artifact_storage_dir: directory for storing artifacts
+    :param customized_annotation_folder: directory for storing customized annotations
+    :param do_not_track: opt out or user analytics
+    :param logging_level: logging level
+    :param logging_file: logging file location (only support local for at this time)
+    :param storage_options: a dictionary for artifact storage configuration(same as storage_options in pandas, Dask and fsspec)
     """
 
     home_dir: Path
     database_url: Optional[str]
-    artifact_storage_dir: Optional[Path]
-    customized_annotation_folder: Optional[Path]
+    artifact_storage_dir: Optional[FilePath]
+    customized_annotation_folder: Optional[FilePath]
     do_not_track: bool
     logging_level: str
-    logging_file: Optional[PathLike]
+    logging_file: Optional[Path]
+    storage_options: Optional[Dict[str, Any]]
 
     def __init__(
         self,
-        home_dir=f"{os.environ.get('HOME','~')}/{LINEAPY_FOLDER_NAME}",
+        home_dir=f"{Path(os.environ.get('HOME','~')).expanduser().resolve()}/{LINEAPY_FOLDER_NAME}",
         database_url=None,
         artifact_storage_dir=None,
         customized_annotation_folder=None,
         do_not_track=False,
         logging_level="INFO",
         logging_file=None,
+        storage_options=None,
     ):
         if logging_level.isdigit():
             logging_level = logging._levelToName[int(logging_level)]
 
-        self.home_dir = home_dir
+        self.home_dir = Path(home_dir).expanduser()
         self.database_url = database_url
         self.artifact_storage_dir = artifact_storage_dir
```

Comment:
should we send a heartbeat here as well as we do with the set call for key "artifact_storage_dir"?